vm/src/builtins/float.rs

```diff
@@ -552,39 +553,49 @@ impl Hashable for PyFloat {
     }
 }
 
-impl AsNumber for PyFloat {
-    const AS_NUMBER: PyNumberMethods = PyNumberMethods {
-        add: Some(|number, other, vm| Self::number_float_op(number, other, |a, b| a + b, vm)),
-        subtract: Some(|number, other, vm| Self::number_float_op(number, other, |a, b| a - b, vm)),
-        multiply: Some(|number, other, vm| Self::number_float_op(number, other, |a, b| a * b, vm)),
-        remainder: Some(|number, other, vm| Self::number_general_op(number, other, inner_mod, vm)),
-        divmod: Some(|number, other, vm| Self::number_general_op(number, other, inner_divmod, vm)),
-        power: Some(|number, other, vm| Self::number_general_op(number, other, float_pow, vm)),
-        negative: Some(|number, vm| {
-            let value = Self::number_downcast(number).value;
-            (-value).to_pyresult(vm)
-        }),
-        positive: Some(|number, vm| Self::number_float(number, vm).to_pyresult(vm)),
-        absolute: Some(|number, vm| {
-            let value = Self::number_downcast(number).value;
-            value.abs().to_pyresult(vm)
-        }),
-        boolean: Some(|number, _vm| Ok(Self::number_downcast(number).value.is_zero())),
-        int: Some(|number, vm| {
-            let value = Self::number_downcast(number).value;
-            try_to_bigint(value, vm).map(|x| vm.ctx.new_int(x))
-        }),
-        float: Some(|number, vm| Ok(Self::number_float(number, vm))),
-        floor_divide: Some(|number, other, vm| {
-            Self::number_general_op(number, other, inner_floordiv, vm)
-        }),
-        true_divide: Some(|number, other, vm| {
-            Self::number_general_op(number, other, inner_div, vm)
-        }),
-        ..PyNumberMethods::NOT_IMPLEMENTED
+macro_rules! atomic_func {
+    ($x:expr) => {
+        AtomicCell::new(Some($x))
     };
 }
 
+impl AsNumber for PyFloat {
+    fn as_number() -> &'static PyNumberMethods {
+        &AS_NUMBER
+    }
+}
+
+static AS_NUMBER: PyNumberMethods = PyNumberMethods {
```

Comment:
even if we make this out of trait, it still can be defined in `fn as_number()` method unless we need it as a pub